pyomo/gdp/plugins/bigm.py

```diff
@@ -378,41 +384,28 @@ def _bigM_relax_disjunct(self, obj, transBlock, bigM, suffix_list):
                         "The disjunct %s is deactivated, but the "
                         "indicator_var is fixed to %s. This makes no sense."
                         % ( obj.name, value(obj.indicator_var) ))
-            if not infodict.get('relaxed', False):
+            if obj._transformation_block is None:
                 raise GDP_Error(
                     "The disjunct %s is deactivated, but the "
                     "indicator_var is not fixed and the disjunct does not "
-                    "appear to have been relaxed. This makes no sense."
+                    "appear to have been relaxed. This makes no sense. "
+                    "(If the intent is to deactivate the disjunct, fix its "
+                    "indicator_var to 0.)"
                     % ( obj.name, ))
-
-        if 'bigm' in infodict:
-            # we've transformed it (with BigM), so don't do it again.
-            return
-
-        # add reference to original disjunct to info dict on transformation
-        # block
+                
+        if obj._transformation_block is not None:
+            # we've transformed it, which means this is the second time it's
+            # appearing in a Disjunction
+            raise GDP_Error(
+                    "The disjunct %s has been transformed, but a disjunction "
+                    "it appears in has not. Putting the same disjunct in "
+                    "multiple disjunctions is not supported." % obj.name)
```

Comment:
Oh, interesting. There's not, but it wouldn't be hard to do... I'm not sure how we want to tell the transformation that this is what we want because I don't think we'd want the default to be for it to transform orphan Disjuncts. And even if you specified a Disjunct in the list of targets, how do you distinguish between treating a Disjunct target as a Block (transform everything on it) vs. transforming it without it being part of a Disjunction? But in any case, the actual transformation would be easy to do.